### ISSUE TITLE
feature request: query cluster intialization status

### ISSUE THREAD

**Issue by lambdai** [envoyproxy/envoy#15178]
We can now query the active cluster and warm cluster to know what clusters are able to use. 
It's adequate to explain why a request is "NR 503", but it's still hard to troubleshooting why a cluster stays warming up.

We can expose the per cluster init::manager to cluster manager and query init manager's unready target.

The followings are need to be resolved.
1. The cluster ready details query should be exposed to admin interface.
2. The current cluster initialization does not end with init::manager announce "ready". 

Thoughts?

**Comment by mattklein123** [comment]
I agree this would be very useful. I would propose some kind of `/init_status` endpoint which gives details on the init process?

**Comment by lambdai** [comment]
We have `/init_dump`. Is it the `/init_status` you are referring to?

**Comment by mattklein123** [comment]
Sure if we have an existing endpoint we can use it.

**Comment by lambdai** [comment]
Rephrased my proposal: 
Attach each async initialization to a named target.
Independently expose the per warming cluster init::manager to the existing `init_dump`
